platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -177,23 +177,19 @@ class ProblemReportingCrossProjectModelAccess(
             return delegate.hashCode()
         }
 
-        override fun getProperty(propertyName: String): Any {
+        override fun getProperty(propertyName: String): Any? {
             // Attempt to get the property value via this instance. If not present, then attempt to lookup via the delegate
             val thisBean = BeanDynamicObject(this).withNotImplementsMissing()
             val result = thisBean.tryGetProperty(propertyName)
             if (result.isFound) {
                 return result.value
             }
             onAccess()
-            val delegateBean = (delegate as DynamicObjectAware).asDynamicObject
-            val delegateResult = delegateBean.tryGetProperty(propertyName)
-            if (delegateResult.isFound) {
-                return delegateResult.value
-            }
-            throw thisBean.getMissingProperty(propertyName)
+
+            return (delegate as DefaultProject).getProperty(propertyName)
```

Comment:
❓ Given that these two implementations assume `delegate` to be a `DefaultProject`, would it make sense to change the property type altogether?